server/src/main/java/de/uftos/repositories/ucdl/parser/DefinitionParser.java

```diff
@@ -304,14 +327,33 @@ private static AbstractSyntaxTreeDto buildAst(Node root, HashMap<String, Resourc
                 AbstractSyntaxTreeDto ast;
 
                 List<AbstractSyntaxTreeDto> params = new ArrayList<>();
-                ElementDto element = (ElementDto) buildAst(root.jjtGetChild(0), parameters);
+                AbstractSyntaxTreeDto element = buildAst(root.jjtGetChild(0), parameters);
+                ResourceType elementType;
+                if (element.getToken() == UcdlToken.NUMBER
+                    || element.getToken() == UcdlToken.SIZE) {
+                  elementType = ResourceType.NUMBER;
+                } else {
+                  elementType = ((ElementDto) element).type();
+                }
```

Comment:
This and the if below for sets are used multiple times across the code. I think it makes sense to introduce a `getType` and `getSetType` private function in here